receiver.go

```diff
@@ -0,0 +1,229 @@
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.VarcharArrayOID:
+		val := r.Val.(*pgtype.VarcharArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.Int8ArrayOID:
+		val := r.Val.(*pgtype.Int8Array)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.Float4ArrayOID:
+		val := r.Val.(*pgtype.Float4Array)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.Float8ArrayOID:
+		val := r.Val.(*pgtype.Float8Array)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.ACLItemArrayOID:
+		val := r.Val.(*pgtype.ACLItemArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.InetArrayOID:
+		val := r.Val.(*pgtype.InetArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.TimestampArrayOID:
+		val := r.Val.(*pgtype.TimestampArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.DateArrayOID:
+		val := r.Val.(*pgtype.DateArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.TimestamptzArrayOID:
+		val := r.Val.(*pgtype.TimestamptzArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.NumericArrayOID:
+		val := r.Val.(*pgtype.NumericArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.UUIDArrayOID:
+		val := r.Val.(*pgtype.UUIDArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.JSONBArrayOID:
+		val := r.Val.(*pgtype.JSONBArray)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return []interface{}{}
+		default:
+			return val.Elements
+		}
+	case pgtype.NameOID:
+		val := r.Val.(*pgtype.Name)
+		switch val.Status {
+		case pgtype.Null:
+			return nil
+		case pgtype.Undefined:
+			return ``
+		default:
+			return val.String
+		}
+	default:
+		return r.Val
+	}
```

Comment:
The type you're suggesting `pgtype.ValueArray` doesn't exist. I considered writing a generic function to do this but that doesn't really work for non-interfaces to my understanding. Can you make a better suggestion or is this probably fine as is?

Reply:
Since I don't control the pgtype library I can't easily add the methods you're suggesting without re-implementing every type (which in the end would be even more duplicated code).